hetznerrobot/provider.go

```diff
@@ -2,38 +2,39 @@ package hetznerrobot
 
 import (
 	"context"
+
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/lenstra/hetzner"
 )
 
 // Provider -
 func Provider() *schema.Provider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"username": &schema.Schema{
-				Type: schema.TypeString,
-				Optional: true,
+			"username": {
+				Type:        schema.TypeString,
+				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("HETZNERROBOT_USERNAME", nil),
 			},
-			"password": &schema.Schema{
-				Type: schema.TypeString,
-				Optional: true,
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("HETZNERROBOT_PASSWORD", nil),
 			},
 		},
-		ResourcesMap:   map[string]*schema.Resource{
+		ResourcesMap: map[string]*schema.Resource{
 			"hetznerrobot_firewall": resourceFirewall(),
 		},
-		DataSourcesMap: map[string]*schema.Resource{},
+		DataSourcesMap:       map[string]*schema.Resource{},
 		ConfigureContextFunc: providerConfigure,
 	}
 }
 
 func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
-	username := d.Get("username").(string)
-	password := d.Get("password").(string)
-
-	var diags diag.Diagnostics
+	config := hetzner.DefaultConfig()
+	config.User = d.Get("username").(string)
+	config.Password = d.Get("password").(string)
```

Comment:
I think it would be very handy to configure a logger here, but doing something like 
```
config.Logger = hclog.Default()
// There is no convenient way to get the configured Terraform logging level, so just always
// emit the Debug logs. They will only appear if the user has enabled logging.
config.Logger.SetLevel(hclog.Debug)
```
Will display all the helpful diagnostic logs from the client library when logging is enabled via https://www.terraform.io/docs/extend/debugging.html#turning-on-logging